Subject: [PATCH] Notify battery stats service when supplicant is killed

Battery History details indicate that Wi-Fi is on when
both "Wi-Fi scanning" and Wi-Fi is disabled. This is due to
battery stats service is not informed that the supplicant has
died.

When supplicant state is reset inform battery stats service
and broadcast receivers that state has changed to UNITIALIZED.

diff --git a/service/java/com/android/server/wifi/SupplicantStateTracker.java b/service/java/com/android/server/wifi/SupplicantStateTracker.java
index ea5d9eb..7b86b88 100644
--- a/service/java/com/android/server/wifi/SupplicantStateTracker.java
+++ b/service/java/com/android/server/wifi/SupplicantStateTracker.java
@@ -230,6 +230,12 @@
                     transitionOnSupplicantStateChange(stateChangeResult);
                     break;
                 case WifiStateMachine.CMD_RESET_SUPPLICANT_STATE:
+                    try {
+                        mBatteryStats.noteWifiSupplicantStateChanged(
+                            BatteryStats.WIFI_SUPPL_STATE_UNINITIALIZED, false);
+                    } catch (RemoteException e) {
+                        // Won't happen.
+                    }
                     transitionTo(mUninitializedState);
                     break;
                 case WifiManager.CONNECT_NETWORK:
